egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,352 @@
+                    snrs_added.append(background_snrs.next())
+                    start_times_added.append(0)
+
+            # Add the point-source noise
+            if len(noise_list) > 0 and random.random() < noise_adding_probability:
+                for k in range(random.randint(1, max_noises_recording)):
+                    # pick the RIR to reverberate the point-source noise
+                    noise = PickItemWithProbability(noise_list)
+                    noise_rir = PickItemWithProbability(room.rir_list)
+                    if noise.bg_fg_type == "background": 
+                        start_times_added.append(0)
+                        noises_added.append("wav-reverberate --duration={2} --impulse-response={1} {0} - |".format(noise.noise_file_location, noise_rir.rir_file_location, speech_dur))
+                        snrs_added.append(background_snrs.next())
+                    else:
+                        start_times_added.append(round(random.random() * speech_dur, 2))
+                        noises_added.append("wav-reverberate --impulse-response={1} {0} - |".format(noise.noise_file_location, noise_rir.rir_file_location))
+                        snrs_added.append(foreground_snrs.next())
+
+            if len(noises_added) > 0:
+                command_opts += "--additive-signals='{0}' ".format(','.join(noises_added))
+            if len(snrs_added) > 0:
+                command_opts += "--snrs='{0}' ".format(','.join(map(lambda x:str(x),snrs_added)))
+            if len(start_times_added) > 0:
+                command_opts += "--start-times='{0}' ".format(','.join(map(lambda x:str(x),start_times_added)))
+            
+            if command_opts == "":
+                command = "{0} {1}\n".format(wav_id, wav_pipe) 
+            else:
+                command = "{0} {1} wav-reverberate {2} - - |\n".format(wav_id, wav_pipe, command_opts)
+
+            command_list.append(command)
+
+    file_handle = open(output_dir + "/wav.scp", 'w')
+    file_handle.write("".join(command_list))
+    file_handle.close()
+
+
+# This function replicate the entries in files like segments, utt2spk, text
+def AddPrefixToFields(input_file, output_file, num_replicas, prefix, field = [0]):
+    list = map(lambda x: x.strip(), open(input_file))
+    f = open(output_file, "w")
+    for i in range(num_replicas):
+        for line in list:
+            if len(line) > 0 and line[0] != ';':
+                split1 = line.split()
+                for j in field:
+                    if prefix is not None:
+                        split1[j] = prefix + str(i) + "_" + split1[j]
+                print(" ".join(split1), file=f)
+            else:
+                print(line, file=f)
+    f.close()
+
+
+def CreateReverberatedCopy(input_dir, output_dir, room_dict, noise_list, foreground_snr_string, background_snr_string, num_replicas, prefix, speech_rvb_probability, noise_adding_probability, max_noises_per_minute):
```

Comment:
Add a comment describing this function. Have one line per argument and describe each argument (and its format) in detail.